Application/EdFi.Ods.Api/Security/Authorization/AuthorizationSegmentSqlProviderBase.cs

```diff
@@ -45,100 +46,79 @@ public AuthorizationSegmentSqlProviderBase(IAuthorizationViewsProvider authoriza
         {
             var segmentStatements = new List<string>();
             var parameters = new List<DbParameter>();
+            var unsupportedAuthorizationViews = new List<string>();
+            var edOrgTypes = new List<string>();
 
             foreach (var authorizationSegment in authorizationSegments)
             {
-                // Within each claim segment, group the values by ed org type, and combine with "OR"
-                var claimEndpointsGroupedByName = authorizationSegment.ClaimsEndpoints
-                    .GroupBy(ep => ep.Name)
-                    .Select(g => g)
-                    .ToList();
+                string claimEndpointName = EducationOrganizationEntitySpecification.IsEducationOrganizationIdentifier(authorizationSegment.ClaimsEndpoints.FirstOrDefault().Name)
```

Comment:
Don't use ternary operator to replace and compact what should just be a conditional check -- it makes the code harder to understand. Use a separate `if` statement here to determine if the claim endpoint is not an EdOrg Identifier, and then throw an exception.

Then to emphasize that `claimEndpointName` is to carry a fixed value of "EducationOrganizationId", declare it as a constant if we're still here (i.e. if no exception is thrown).

Reply:
Fixed it